client/src/app/shared/execution-card/execution-card.component.ts

```diff
@@ -1,17 +1,26 @@
-import { Component, Input, Output, EventEmitter } from '@angular/core';
-
+import { Component, Input, Output, EventEmitter, OnChanges, SimpleChanges } from '@angular/core';
 import { Execution } from '../../models/execution';
 import { ExecutionStatus } from '@machinelabs/models';
+import { OutputFilesService } from '../../output-files.service';
+import { Observable } from 'rxjs';
 
 @Component({
   selector: 'ml-execution-card',
   templateUrl: './execution-card.component.html',
   styleUrls: ['./execution-card.component.scss']
 })
-export class ExecutionCardComponent {
+export class ExecutionCardComponent implements OnChanges {
+  hasOutput$: Observable<boolean>;
+
+  constructor(public outputFilesService: OutputFilesService) {}
+
   @Input() execution;
 
   @Output() view = new EventEmitter<Execution>();
-
   ExecutionStatus = ExecutionStatus;
+  ngOnChanges(simple: SimpleChanges): void {
+    if (simple.hasOwnProperty('execution') && simple.execution.currentValue !== null) {
```

Comment:
I don't think we need to use `hasOwnProperty()` here as we don't expect `execution` to be on the prototype anyways.